core/control-plane/contract-core/src/main/java/org/eclipse/edc/connector/contract/offer/ContractOfferResolverImpl.java

```diff
@@ -118,22 +119,25 @@ private Stream<ContractOffer.Builder> createContractOffers(ContractDefinition de
     @NotNull
     private ContractOffer.Builder createContractOffer(ContractDefinition definition, Policy policy, Asset asset) {
 
-        var contractEndTime = calculateContractEnd(definition);
+        var start = clock.instant();
+        var zone = clock.getZone();
+        var contractEndTime = calculateContractEnd(definition, start, zone);
+        var contractStartTime = ZonedDateTime.ofInstant(start, zone);
 
         return ContractOffer.Builder.newInstance()
                 .id(ContractId.createContractId(definition.getId()))
                 .policy(policy.withTarget(asset.getId()))
                 .asset(asset)
-                .contractStart(ZonedDateTime.now())
+                .contractStart(contractStartTime)
                 .contractEnd(contractEndTime);
     }
 
     @NotNull
-    private ZonedDateTime calculateContractEnd(ContractDefinition definition) {
+    private ZonedDateTime calculateContractEnd(ContractDefinition definition, Instant start, ZoneId zone) {
```

Comment:
This method could work only with `Instant` and leave the `zone` application to the caller.
The method definition would become:
```java 
private Instant calculateContractEnd(ContractDefinition definition, Instant start)
```

Reply:
I could go either way on this. The current method returns a ZonedDateTime, which would keep the change smaller but returning Instant centralize the handling of the time zone.

Reply:
@tuncaytunc-zf can you implement the two changes from @ndr-brt so we can merge this?